nodejs/aws-infra/experimental/cluster.ts

```diff
@@ -0,0 +1,132 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as mod from ".";
+
+import { Network } from "./../network";
+
+import * as utils from "../utils";
+
+/**
+ * A Cluster is a general purpose ECS cluster configured to run in a provided Network.
+ */
+export class Cluster extends pulumi.ComponentResource {
```

Comment:
Decision: Even though there is a aws resource with thsi name, we use the same name.  This is in our 'components' package.  And it helps immediately make clear the general purpose for this type.

Reply:
Did you consider `EcsCluster`, so that someone doesn't confuse this with an EKS cluster?

Reply:
Related to this, would you envision supporting an `EksCluster` in here someday down the road (once it stabilizes a bit)?

Reply:
We could def do EcsCluster.  That's probably a good idea, unless we want this guy itself to live in an 'ecs' module.  will discuss with @lukehoban

Reply:
resolution: try to propose a module structure.